[Fix] Making a nice responsive on every page

JulesGuesnon opened the pull request:
# Description
This PR fixes the responsive on most the pages. I would need someone to test it too, to make sure I didn't miss anything. Also this PR should be merged after #60 is merged, so I can make sure the responsive isn't broken on the new changes

aguillon commented:
At Jules' request and in order to become more familiar with the project, I've tested the UI a bit. Overall, the UI seems pretty good, with minor tweaks that could be fixed in this PR as well IMO.

I've tested the responsiveness during the creation of a wallet, the creation of a proposal and the confirmation of a proposal. It looks good to me in width. I guess the UI could be more responsive in height too (see following screenshot) but I don't think it matters that much.
![responsiveness1](https://user-images.githubusercontent.com/7795522/230427198-59fb365d-8649-46d4-9d89-f9eaa964e845.png)

Some error messages could be improved: 
- "already exists" when adding an account that's already listed, I suggest "This account is already participating" or "Please enter each account only once".
- "invalid days" when entering "0" as days: what if we want proposals that last less than 1 day? I suggest "Proposals must be put to vote for at least 1 day"

This PR does not fix the issue with mutez.

I've tested on mobile as well and I think there's a bit of room for improvement (screenshots from Firefox for Android):
- I've arrived on the expanded sidebar view by accident and was a bit confused on how I should reduce the bar. Perhaps the "← Close sidebar" option should be more visible? ![Screenshot_20230406-174504_Firefox](https://user-images.githubusercontent.com/7795522/230431233-4f5026af-1122-42e9-93fb-3f8e84bd7508.png)
- In horizontal mode, the UI seems unnecessarily stacked: 
![Screenshot_20230406-174612_Firefox](https://user-images.githubusercontent.com/7795522/230431867-282ae27d-c680-4dd8-8619-9b4e6186964e.png)
![Screenshot_20230406-174626_Firefox](https://user-images.githubusercontent.com/7795522/230431872-f46e7d6b-cbe3-4d70-8277-f8b0902b24e8.png)

aguillon reviewed (changes requested):
No opinion on the code.

Please change the two error messages I listed (quick to fix) as well as the "Close sidebar" message, if you agree.

JulesGuesnon commented:
- Updated the error messages
- Brighten the text color of `open sidebar` and `close sidebar`
- Fixed the issue of `Invalid days` now 0 is a correct value, but if every values are 0 there's an error

aguillon commented:
> Fixed the issue of Invalid days now 0 is a correct value, but if every values are 0 there's an error

I think the minimum should be something like 15 minutes or even 1 hour, though. If users set a limit so low that proposals expire almost immediately after being posted, that's an issue as well. I actually can't seem to fix this, probably because my computer is a bit out of sync.

>  Brighten the text color of open sidebar and close sidebar

That's better, thanks. Another issue: when testing on mobile (Firefox for Android), clicking "open sidebar" does nothing, even after associating a wallet. I don't have this issue on Firefox for Linux (even with a small page).

Another issue: when I'm on the "Settings" page and I refresh the page, I'm taken back to the "Proposals" page. Actually it happens with from other pages as well, but there's little reason to refresh other pages I think — whereas Settings can be refreshed when the user wants to see the new settings.

Good luck!

rueshyna commented:
1. We don't force user to add an alias/address.
![image](https://user-images.githubusercontent.com/115862/230857508-05e28540-dca4-4685-a854-da4e9d71c985.png)

2. For Navbar,
- I'd say "address book", "new wallet", "import wallet" should be right align.  
- remove the red avatar circle nearby the account, we don't have the circle in desktop version.   
- I'd say it's a little troublesome to click twice to see "sign out" option. would be nice to directly display with  "address book", "new wallet", "import wallet".
- would it be a good idea that the account(tzx) and network inform should should display directly in Navbar so users don't need to have a one click to check.
![image](https://user-images.githubusercontent.com/115862/230859818-01e5e30d-cd6e-4b57-8cbb-57466f4c0fab.png)

3. close menu from Navbar when open sidebar.
![image](https://user-images.githubusercontent.com/115862/230864781-b60bc242-5503-4d84-bd47-222c4d45c21c.png)

4. for ProposalCard,
- Param -> Param/Tokens
- the texts are overlap. would it be easy if we display like this in mobile version?
- [nice to change] I think if yes/no selection replace with checkbox would be easier for user.
```
Content
Function   Transfer
Metadata   -
Amount     - 
Address    KTabc...efghi
Entrypoint abcdefg
Param/Token      "param"

Function  Execute Contract
Metadata   -
Amount     - 
Address    KTabc...efghi
Entrypoint abcdefg
Param/Token      "param"


Function  Execute Contract
Metadata   -
Amount     - 
Address    KTabc...efghi
Entrypoint abcdefg
Param/Token      "param"

Active 
# same for active
```
![image](https://user-images.githubusercontent.com/115862/230866738-4683a9cc-bf73-4d28-9742-69e073ca53f4.png)
![image](https://user-images.githubusercontent.com/115862/230868332-ed67eff1-f7f4-4142-990f-0e9cffc2c48f.png)


5. when no wallet is imported, there is no sidebar to open but it still shows "Open sidebar ->"
![image](https://user-images.githubusercontent.com/115862/230870225-13e757fe-c764-4887-8db5-3f863161d704.png)

rueshyna commented:
> Another issue: when I'm on the "Settings" page and I refresh the page, I'm taken back to the "Proposals" page. Actually it happens with from other pages as well, but there's little reason to refresh other pages I think — whereas Settings can be refreshed when the user wants to see the new settings.

@JulesGuesnon 
If it's easy to make a change here, then please proceed. otherwise, I think it can come with routing feature, which should come in this quarter. 

> I think the minimum should be something like 15 minutes or even 1 hour, though.

I'd like put some warning messages for it but not force user to follow. 
warning message should warn users when
- proposal duration is low.
- user remove itself from owners.

> If users set a limit so low that proposals expire almost immediately after being posted, that's an issue as well. I actually can't seem to fix this, probably because my computer is a bit out of sync.

It does have some problem with sync issue when things happen too quick. @JulesGuesnon  can you check if it's a quick fix? If not, please create an issue and we can prioritize fix later.

JulesGuesnon commented:
> If it's easy to make a change here, then please proceed. otherwise, I think it can come with routing feature, which should come in this quarter.

I know what's happening, but yeah I'll include a fix with the new routes, now it's a bit off topic I think

I agree with the rest, and I'll include new warnings and check for the sync issue

aguillon commented:
I don't have much time for reviews this week and I see that @rueshyna is back, so I'll skip reviews from now on. :wave:

rueshyna commented:
1. when signed out, the text `sign out` should be hide.
![image](https://user-images.githubusercontent.com/115862/231369152-ea7fb541-386c-4ad9-8a93-fbd2e62d2c07.png)

2. To align with other account presentation, I think the account should display 5 chars in both head and tail.
![image](https://user-images.githubusercontent.com/115862/231370323-1e0811b1-acbb-4f87-bb77-230473341b99.png)

3. the date in active is too closed to account, especially, the hyphen.
![image](https://user-images.githubusercontent.com/115862/231370734-ba06230e-a6f1-40c0-8ea7-ea974241f60a.png)

4. should be Params -> Params/Token everywhere.
![image](https://user-images.githubusercontent.com/115862/231371235-351336d9-1343-44bd-b3cf-92c4e8d1eadb.png)

rueshyna commented on components/signersForm.tsx:
```diff
@@ -552,9 +567,14 @@ const SignersForm: FC<{
+                  renderError(errors.proposalDuration)
+                : currentDuration < PROPOSAL_DURATION_WARNING
+                ? renderWarning(
+                    "Proposal duration is low, you may not be able to resolve proposals"
```
when proposal is expired, users still can resolve the proposals. Therefore, I suggest s/resolve/execute.

rueshyna commented on pages/address-book.tsx:
```diff
@@ -111,130 +108,148 @@ function Home() {
+                    errors.validators as
+                      | { name: string; address: string }[]
+                      | undefined
+                  )?.every(v => v.address === "" && v.name === "") ??
```
when click "Add an address" button, 
```
Unhandled Runtime Error
TypeError: Cannot read properties of null (reading 'address')

Source
pages/address-book.tsx (117:34) @ address

  115 |     | { name: string; address: string }[]
  116 |     | undefined
> 117 | )?.every(v => v.address === "" && v.name === "") ??
```

rueshyna commented on components/ProposalCard.tsx:
```diff
@@ -127,7 +127,7 @@ export const RenderProposalContent = ({
         }),
       };
-    } else {
+    } else if (metadata.meta) {
```
In which case, we will have `meta` in `metadata`?

JulesGuesnon replied:
Tbh I don't know. I had a crash with one of my wallets, so I just logged the data and handled the case to avoid crashes. It may be some legacy of the previous contract version
